### PR TITLE
Shader fix, engine settings precedence, "Force reimport" context menu option

### PR DESCRIPTION
**Pull request by mattbenic**

### CODE REVIEW

**Comment by JoshEngebretson** [comment]
Awesome!  This is looking solid.  One thing I want to look into is the "--fromeditorplay" flag and "runningFromEditorPlay_" variable (this could be a followup issue).  I'll do a quick code review on this tomorrow, THANKS!

**Comment by mattbenic** [comment]
That was for distinguishing that the editor is running in player mode, but from the editor as a separate process. It still feels like there needs to be some kind of mode enum that distinguishes code running in editor, editor player from the editor, editor player standalone, and built player standalone.

File: AUTHORS.md [hunk 1]
```diff
@@ -23,6 +23,10 @@
 
 - Nick Minkler (https://github.com/Sleaker)
 
+- Raheel Hassim (https://github.com/raheelx)
+
+- Gareth Fouche (https://github.com/GarethNN)
+
```
Comment (JoshEngebretson) [inline AUTHORS.md]: Awesome! :+1:

File: Resources/CoreData/Shaders/HLSL/LitSolid.hlsl [hunk 2]
```diff
@@ -310,11 +312,16 @@ void PS(
...
         #ifdef LIGHTMAP
             finalColor += Sample2D(EmissiveMap, iTexCoord2).rgb * diffColor.rgb;
+			
+			#ifdef ENVCUBEMAP
+				finalColor += cMatEnvMapColor * SampleCube(EnvCubeMap, reflect(iReflectionVec, normal)).rgb * Sample2D(EmissiveMap, iTexCoord2).rgb;
+			#endif
+		#else
+			#ifdef ENVCUBEMAP
+				finalColor += cMatEnvMapColor * SampleCube(EnvCubeMap, reflect(iReflectionVec, normal)).rgb;
+			#endif
```
Comment (JoshEngebretson) [inline Resources/CoreData/Shaders/HLSL/LitSolid.hlsl]: Whitespace issues, though don't worry, we'll get this sorted in post (ie. have a "system" for handling whitespace here)

File: Script/AtomicEditor/ui/frames/menus/ProjectFrameMenu.ts [hunk 3]
```diff
@@ -79,6 +79,13 @@ class ProjectFrameMenus extends Atomic.ScriptObject {
                 utils.revealInFinder(path);
                 return true;
             }
+
+            if (refid == "force_reimport") {
+                asset.setDirty(true);
+                ToolCore.assetDatabase.scan();
+                return true;
```
Comment (JoshEngebretson) [inline Script/AtomicEditor/ui/frames/menus/ProjectFrameMenu.ts]: Wow, pretty tight!

File: Source/Atomic/Engine/EngineConfig.cpp [hunk 4]
```diff
@@ -326,13 +326,24 @@ bool EngineConfig::LoadFromFile(Context *context, const String& filename)
...
+    }
+    else
     {
-        settings.InsertNew(itr->first_, itr->second_);
-        itr++;
+        while (itr != engineConfig_.End())
+        {
+            settings.InsertNew(itr->first_, itr->second_);
+            itr++;
+        }
```
Comment (JoshEngebretson) [inline Source/Atomic/Engine/EngineConfig.cpp]: Cool, we do need to sort EngineConfig more as we discussed... nice add

**Comment by JoshEngebretson** [comment]
Ok, this looks great, I put up a PR to the Luma master branch with a little cleanup when detecting that we're launching from the editor during setup (before the AEPlayerMode subsystem exists):

https://github.com/LumaDigital/AtomicGameEngine/pull/1

So, if that works for you, when you merge should update here :rocket:

**Comment by mattbenic** [comment]
Cool, will probably only get to that tomorrow. What's your whitespace tool? I'd like to try something this side. I posted something on gitter that might help a bit.

**Comment by JoshEngebretson** [comment]
Awesome, thanks! LANDING! :airplane:

**Comment by JoshEngebretson** [comment]
I think we'll have to come up with something for the HLSL/GLSL on whitespace, I just didn't want to hold up this PR as know we'll need to sort it.  EditorConfig may help, good find

**Comment by mattbenic** [comment]
Also something to check out, your json module definitions are all tabbed right now. Kind of stands out as the only non-third-party case ;)

**Pull request merged**